Cv files are uploaded

gabichelsea opened the pull request:
Trello card link:
[https://trello.com/c/vdKazZQB](https://trello.com/c/vdKazZQB)

gabichelsea committed:
Added File and Folder handler classes
Removed CandidateCvFileHandler util class

gabichelsea committed:
Added getByEmail and getByPhone methods() to CandidateService

gabichelsea committed:
Added pre-validation candidate and file

gabichelsea committed:
Added cvFilename field to CandidateOneWayConverter
Updated FileDownloadController
Updated file upload-download controller tests

gabichelsea committed:
Refactoring atsy-candidate-create.js(ajax)
in validate() methods send error msg instead of error code

gabichelsea committed:
Replaced input 'button' to fileUpload icon

gabichelsea committed:
Added candidateValidatorHelper which validate Candidates

gabichelsea committed:
Added validate() method for StateHistoryController
Refactoring FileUploadController
Added file uploading chance to atsy-statehistory-create.js

gabichelsea committed:
Fixed it CandidateValidatorHelper class
Do restController from FileDownloadController
Added validate() method to FileDownloadController

gabichelsea committed:
Finished:
application_state.jsp
candidate_create.jsp
atsy-candidate-create.js
atsy-statehistory-create.js

gabichelsea committed:
Tested:
findByPhone, findByEmail, getCandidateDtoByEmail, getCandidateDtoByPhone

gabichelsea committed:
Refactoring CandidateValidatorHelper class
Added test cases to CandidateValidatorHelper class

gabichelsea committed:
Added CANDIDATE_OBJECT_KEY testing to ApplicationStateControllerTest
Fixed FileDownloadControllerTest.java
Moved download link closer to the label

gabichelsea committed:
Test:
validateStateHistoryViewRepresentation() method in StateHistoryController

gabichelsea committed:
Test:
validateCandidate() method in SingleCandidateController

gabichelsea committed:
Fixed it CandidateOneWayConverterTest
Removed custom message key(when creationDate is null) from StateHistoryViewRepresentation

gabichelsea committed:
Test:
FileDownloadController

gabichelsea committed:
Test:
FileUploadController

Added a new messageKey-exception pair to ExceptionMapperConfiguration

gabichelsea committed:
Removed unnecessary catch block from FileUploadController

gabichelsea committed:
Solved the null folder problem

lamfalusy commented on web/src/main/java/com/epam/rft/atsy/web/handler/FileHandler.java:
```diff
@@ -0,0 +1,20 @@
+package com.epam.rft.atsy.web.handler;
+
+
```
Duplicated empty line.

lamfalusy commented on web/src/main/java/com/epam/rft/atsy/web/handler/FolderHandler.java:
```diff
@@ -0,0 +1,31 @@
+package com.epam.rft.atsy.web.handler;
+
+
```
Duplicated empty line.

lamfalusy commented on web/src/main/java/com/epam/rft/atsy/web/handler/FileHandler.java:
```diff
@@ -0,0 +1,20 @@
+@Component
+public class FileHandler {
+
+  public File getFileByParentDirectoryPathAndFolderNameAndFilename(String parentDirectoryPath,
```
Please add a little java doc for this method.

lamfalusy commented on web/src/main/java/com/epam/rft/atsy/web/handler/FolderHandler.java:
```diff
@@ -0,0 +1,31 @@
+@Component
+public class FolderHandler {
+
+  public void createFolderInParentDirectoryPath(String folderName, String parentDirectoryPath)
```
Please add a little java doc for this method.

lamfalusy commented on web/src/main/java/com/epam/rft/atsy/web/handler/FolderHandler.java:
```diff
@@ -0,0 +1,31 @@
+    FileUtils.forceMkdir(new File(parentDirectoryPath + File.separator + folderName));
+  }
+
+  public boolean existFolderNameInParentDirectoryPath(String folderName, String parentDirectoryPath)
```
Please add a little java doc for this method.

lamfalusy commented on web/src/test/java/com/epam/rft/atsy/web/handler/FileHandlerTest.java:
```diff
@@ -0,0 +1,46 @@
+package com.epam.rft.atsy.web.handler;
+
+
```
Duplicated empty line.

lamfalusy commented on web/src/test/java/com/epam/rft/atsy/web/handler/FileHandlerTest.java:
```diff
@@ -0,0 +1,46 @@
+
+import java.io.File;
+
+public class FileHandlerTest extends AbstractHandlerTest {
```
Use Given-When-Then comments in all test methods as usual.

lamfalusy commented on web/src/test/java/com/epam/rft/atsy/web/handler/FolderHandlerTest.java:
```diff
@@ -0,0 +1,78 @@
+import java.io.File;
+import java.io.IOException;
+
+public class FolderHandlerTest extends AbstractHandlerTest {
```
Use Given-When-Then comments in all test methods as usual.

lamfalusy commented on web/src/main/java/com/epam/rft/atsy/web/controllers/rest/SingleCandidateController.java:
```diff
@@ -57,16 +60,17 @@ public ResponseEntity saveOrUpdate(@Valid @RequestBody CandidateDTO candidateDTO
-        || candidateService.getCandidateDtoByEmail(candidateDTO.getEmail()) != null) {
-      String errorMessage = messageKeyResolver.resolveMessageOrDefault(CANDIDATE_ERROR_DUPLICATE_MESSAGE_KEY);
-      return new ResponseEntity(new RestResponse(errorMessage), HttpStatus.BAD_REQUEST);
+    // Create a new candidate
```
I am not sure what this comment exactly means.

gabichelsea replied:
Why not?:DD

lamfalusy commented on web/src/main/java/com/epam/rft/atsy/web/helper/CandidateValidatorHelper.java:
```diff
@@ -0,0 +1,58 @@
+package com.epam.rft.atsy.web.helper;
+
+
```
Duplicated empty line.

lamfalusy commented on web/src/main/java/com/epam/rft/atsy/web/helper/CandidateValidatorHelper.java:
```diff
@@ -0,0 +1,58 @@
+import org.springframework.util.Assert;
+
+@Component
+public class CandidateValidatorHelper {
```
Please add javadoc to this class.

lamfalusy commented on web/src/main/webapp/resources/js/atsy-statehistory-create.js:
```diff
@@ -58,6 +64,15 @@ function StateHistoryModel() {
+        return self.fileErrorResponse().errorMessage;
+    });
+
+
```
Duplicated empty line.

lamfalusy commented on web/src/test/java/com/epam/rft/atsy/web/controllers/rest/SingleCandidateControllerTest.java:
```diff
@@ -296,12 +321,217 @@ public void saveOrUpdateShouldRespondWithErrorResponseWhenLanguageSkillIsTooHigh
+    verifyZeroInteractions(this.messageKeyResolver);
+  }
+
+
```
Duplicated empty line.

lamfalusy commented:
There is a private field in com.epam.rft.atsy.web.controllers.rest.FileUploadController called CANDIDATE_ALREADY_HAS_CV_FILE_MESSAGE_KEY, which is never used.

lamfalusy commented:
There is a big hole in the form:
![image](https://cloud.githubusercontent.com/assets/6324917/18877212/ca62f616-84cc-11e6-9808-5d3e45323b4e.png)

gabichelsea committed:
Deleted empty lines
Fixed it "big hole css 'feature'"
Added Given-Then-When comments to Handler tests
